### ISSUE TITLE
Applicative Functor

### ISSUE THREAD

**Issue by kasozivincent** [louthy/language-ext#1252]
Hi everyone. Does this library support applicative functors? I have tired using Apply with a binary function inside an Either on 2 Either values but it doesn't work and there is no document. I need help. Thank you

**Comment by louthy** [comment]
When you report issues, please give more information than "it doesn't work", that doesn't allow me to help you in any way because I have no context.

But if you want to use `Apply` then there are two ways of doing it with a binary function:
```c#
var r3 = Right<string, Func<int, int, int>>(Add)
            .Apply(Right(100))
            .Apply(Right(200));

var r4 = Right<string, Func<int, int, int>>(Add)
            .Apply(Right(100), Right(200));

int Add(int x, int y) => 
    x + y;
```
For synchronous monadic types it's nearly always easier to just use LINQ:
```c#
    var r = from x in Right(100)
            from y in Right(200)
            select x + y;
```
Asynchronous monads and the `Validation` monad benefit from using `Apply` because they can do parallel processing (asynchronous monads) and collection of multiple errors (`Validation`).

**Comment by louthy** [comment]
The right-constructor function `EitherRight<R> Right<R>(R value)` produces an `EitherRight<R>` value which can be implicitly cast to `Either<L, R>`; but I noticed that `EitherRight<R>` didn't support `Apply` (whereas `Either<L, R>` did).

So in `v4.4.6` I have added support for that now.  It removes the need to specify the generics for the initial `Right` (well, most of the time, the compiler isn't always helpful with `MethodInfo → Func` inference).

It makes the above examples a bit more elegant:
```c#
var r1 = Right(Add)
            .Apply(Right(100)).Apply(Right(200));

var r2 = Right(Add)
            .Apply(Right(100), Right(200));
```
